plugins/python/python_plugin_common.c

```diff
@@ -371,6 +375,105 @@ _python_plugin_register_plugin_in_py_ctx(void)
     debug_return_int(SUDO_RC_OK);
 }
 
+int
+_python_plugin_set_path(struct PluginContext *plugin_ctx, const char *path)
+{
+    if (path == NULL) {
+        py_sudo_log(SUDO_CONV_ERROR_MSG, "No python module path is specified. "
+                                         "Use 'ModulePath' plugin config option in 'sudo.conf'\n");
+        return SUDO_RC_ERROR;
+    }
+
+    if (*path == '/') { // absolute path
+        plugin_ctx->plugin_path = strdup(path);
+    } else {
+        asprintf(&plugin_ctx->plugin_path, PLUGIN_DIR "/python/%s", path);
```

Comment:
This is probably OK but asprintf() does not set the pointer to NULL on failure.  I suggest checking the return value and clearing plugin_ctx->plugin_path on failure just to be safe.  E.g.
```
if (asprintf(&plugin_ctx->plugin_path, PLUGIN_DIR "/python/%s", path) == -1)
     plugin_ctx->plugin_path = NULL;
```

Reply:
Fixed it